include/dem/dem.h

```diff
@@ -66,251 +68,465 @@ class DEMSolver
+   * angular kinetic.
    */
   void
   report_statistics();
 
-  MPI_Comm                                  mpi_communicator;
-  const unsigned int                        n_mpi_processes;
-  const unsigned int                        this_mpi_process;
-  ConditionalOStream                        pcout;
-  DEMSolverParameters<dim>                  parameters;
+  /**
+   * @brief Execute the sorting of particle into subdomains and cells, and
+   * reinitialize the containers dependent on the local particle ids.
+   */
+  void
+  sort_particles_into_subdomains_and_cells();
+
+  /**
+   * @brief The MPI communicator used for the parallel simulation.
+   */
+  MPI_Comm mpi_communicator;
+
+  /**
+   * @brief The number of MPI processes used for the parallel simulation.
+   */
+  const unsigned int n_mpi_processes;
+
+  /**
+   * @brief The rank of the current MPI process.
+   */
+  const unsigned int this_mpi_process;
+
+  /**
+   * @brief The output stream used for the parallel simulation, only used by
+   * the process 0.
+   */
+  ConditionalOStream pcout;
+
+  /**
+   * @brief The parameters of the DEM simulation.
+   */
+  DEMSolverParameters<dim> parameters;
+
+  /**
+   * @brief The distributed triangulation used for the DEM simulation.
+   */
   parallel::distributed::Triangulation<dim> triangulation;
 
-  MappingQGeneric<dim>                 mapping;
-  bool                                 particles_insertion_step;
-  unsigned int                         contact_build_number;
-  TimerOutput                          computing_timer;
-  double                               smallest_contact_search_criterion;
-  double                               smallest_floating_mesh_mapping_criterion;
+  /**
+   * @brief The polynomial mapping. The mapping is first order in DEM
+   * simulations.
+   */
+  MappingQGeneric<dim> mapping;
+
+  /**
+   * @brief The particle handler that manages the particles.
+   */
   Particles::ParticleHandler<dim, dim> particle_handler;
-  bool                                 contact_detection_step;
-  bool                                 load_balance_step;
-  bool                                 checkpoint_step;
-  Tensor<1, 3>                         g;
-
-  DEM::DEMProperties<dim>                  properties_class;
-  std::vector<std::pair<std::string, int>> properties =
-    properties_class.get_properties_name();
-  double             neighborhood_threshold_squared;
-  double             maximum_particle_diameter;
+
+  /**
+   * @brief The action manager that manages the actions triggered by events.
+   */
+  DEMActionManager *action_manager;
+
+  /**
+   * @brief The timer that keeps track of the time spent in some functions.
+   * Currently theses functions are: load balancing and VTU output.
+   */
+  TimerOutput computing_timer;
+
+  /**
+   * @brief The properties of the DEM simulation.
+   */
+  DEM::DEMProperties<dim> properties_class;
+
+  /**
+   * @brief The acceleration acting of the particles.
```

Comment:
```suggestion
   * @brief The gravitational acceleration acting on the particles.
```

Reply:
Theoretically, it can be other forces (let's say from a electromagnetic field). Like, we use it for gravity and the letter it g, but stilll!

Reply:
agreed